export.py

```diff
@@ -0,0 +1,251 @@
+  --name NAME, -n NAME  The name to use for saving the exported ONNX model
+##########
+Example usage:
+python export.py --checkpoint ./checkpoints/vit_base_patch32_224-224_pruned.pth.tar \
+    --recipe ./recipes/vit_base.85.recal.config.yaml \
+##########
+Example Two:
+python export.py --checkpoint ./quantized-checkpoint/vit_base_patch32_224-224_pruned.pth.tar \
+    --recipe ./recipes/vit_base.85.quant.config.yaml \
+    --save-dir ./exported-models \
+    --name vit_base_patch32_224-224 \
+    --batch-size 1 \
+    --image-shape 3 550 550 \
+    --config ./quantized-checkpoint/args.yaml
+"""
+
+import os
+import yaml
+import argparse
+from argparse import Namespace
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterable
+import torch
+from timm.models import create_model
+from timm.optim import create_optimizer_v2, optimizer_kwargs
+from sparseml.pytorch.optim import ScheduledModifierManager
+from sparseml.pytorch.utils import export_onnx
+
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclass
+class ExportArgs:
+    """
+    Typed arguments for exporting a ViT model to ONNX
+    """
+
+    checkpoint: Path
+    config: str
+    recipe: str
+    no_qat: bool
+    batch_size: int
+    image_shape: Iterable
+    name: Path
+
+    def __post_init__(self):
+        """
+        post-initialization processing and validation
+        """
+        self.checkpoint = Path(self.checkpoint)
+
+        if not self.checkpoint.exists():
+            raise FileNotFoundError(
+                f"The checkpoint {self.checkpoint} does " f"not exist."
+            )
+
+        self.config = Path(self.config or os.path.dirname(self.checkpoint) + "/args.yaml")
+        
+        if not self.config.exists():
+            raise FileNotFoundError(
+                f"The config file {self.config} does " f"not exist."
+            )
+        
+        self.image_shape = tuple(self.image_shape)
+        name_path = Path(self.name) if self.name else Path(".")
+        if name_path.suffix == '':
+            self.save_dir = name_path
+            self.name = ''
+        else:
+            self.save_dir = name_path.parent
+            self.name = name_path.name
+        self.save_dir.mkdir(parents=True, exist_ok=True)
+        self.name = self.get_safe_name()
+
+    def get_safe_name(self):
+        if self.name:
+            self.name = Path(self.name)
+        else:
+            self.name = Path(self.checkpoint.with_suffix(".onnx").name)
+
+        filename = self.name.stem
+        self.name = self.save_dir / f"{filename}.onnx"
+        existence_counter = 0
+        while self.name.exists():
+            existence_counter += 1
+            self.name = self.name.parent / f"{filename}-{existence_counter}.onnx"
+        return self.name
+
+
+def parse_args() -> ExportArgs:
+    """
+    Add and parse arguments for exporting a ViT model to ONNX
+    :return: A ExportArgs Object with typed arguments
```

Comment:
Sphinx needs an empty line before params to generate docs
```suggestion
    Add and parse arguments for exporting a ViT model to ONNX
    
    :return: A ExportArgs Object with typed arguments
```